cpp/src/arrow/compute/exec/plan_test.cc

```diff
@@ -1016,9 +1019,14 @@ TEST(ExecPlanExecution, ProjectMaintainsOrder) {
   constexpr int kRandomSeed = 42;
   constexpr int64_t kRowsPerBatch = 1;
   constexpr int kNumBatches = 16;
-  auto source_node = gen::Gen({{"x", gen::Step()}})
-                         ->FailOnError()
-                         ->SourceNode(kRowsPerBatch, kNumBatches);
+
+  auto generator = gen::Gen({{"x", gen::Step()}})->FailOnError();
```

Comment:
Curious why this change?

Reply:
This was using generator.h with compute/exec/options.h. I removed that include (and the associated exec options from generator, that is `SourceNode()`) and expanded it in this test, which was the only place where it was used.

Reply:
I am planning on starting to use these utilities in more places.  However, we can introduce `exec::gen::Gen` (or something like that) which wraps the generator and adds exec-specific utilities.  So I think this is tolerable for now.